R/forecastHybrid_model.R

```diff
@@ -0,0 +1,86 @@
+#' forecastHybrid model wrapper
+#'
+#' Allows users to forecast using ensembles from the `forecastHybrid` package. Note that
+#' whilst weighted ensembles can be created this is not advised when samples > 1 as currently
+#' samples are derived assuming a normal distribution using the upper and lower confidence intervals of the ensemble.
+#' These confidence intervals are themselves either based on the unweighted mean of the ensembled
+#' models or the maximum/minimum from the candiate models. Note that `forecastHybrid` must be installed for this
+#' model wrapper to be functional.
+#' @param y Numeric vector of time points to forecast
+#' @param samples Numeric, number of samples to take.
+#' @param horizon Numeric, the time horizon over which to predict.
+#' @param model_params List of parameters to pass to `forecastHybrid::hybridModel`.
+#' @param forecast_params List of parameters to pass to `forecastHybrid:::forecast.hybridModel`.
+#' @return A dataframe of predictions (with columns representing the time horizon and rows representing samples).
+#' @export
+#' @importFrom purrr map2
+#' @importFrom dplyr bind_cols
+#' @examples \dontrun{
+#'
+#' library(forecastHybrid)
+#'
+#' ## Used on its own
+#' forecastHybrid_model(y = EpiSoon::example_obs_rts$rt,
+#'                      samples = 10, horizon = 7)
+#'
+#'
+#'## Used with non-default arguments
+#'## Note that with the current sampling from maximal confidence intervals model
+#'## Weighting using cross-validation will only have an impact whhen 1 sample is used.
+#'forecastHybrid_model(y = EpiSoon::example_obs_rts$rt,
+#'                     samples = 1, horizon = 7,
+#'                     model_params = list(cvHorizon = 7, windowSize = 7,
+#'                                   rolling = TRUE, models = "zta"))
+#'
+#'
+#' ## Used for forecasting
+#'  forecast_rt(EpiSoon::example_obs_rts,
+#'                     model = EpiSoon::forecastHybrid_model,
+#'                     horizon = 7, samples = 1)
+#'
+#'## Used for forcasting with non-default arguments
+#'forecast_rt(EpiSoon::example_obs_rts,
+#'            model = function(...){EpiSoon::forecastHybrid_model(
+#'            model_params = list(models = "zte"),
+#'            forecast_params = list(PI.combination = "mean"), ...)
+#'            },
+#'            horizon = 7, samples = 10)
+#'}
+forecastHybrid_model <- function(y = NULL, samples = NULL,
+                                 horizon = NULL, model_params = NULL,
+                                 forecast_params = NULL) {
+
+
+  check_suggests("forecastHybrid")
+
+
+  ## Fit the model
+  fitted_model <- suppressMessages(
```

Comment:
what are the warnings that get suppressed? I don't get warnings with the example code

Reply:
ah. ok now I got some for one of the models

Reply:
still a bit hacky :D

Reply:
alternative suggests 
we don't want 2000 warnings